server/api/cli/cli.controller.js

```diff
@@ -39,17 +39,22 @@ exports.run = function (req, res) {
   dirName += dateTime.getFullYear()
   const execFile = req.body.fileName.replace('.c', '')
   const cmd = `"${dirName}/${execFile}"`
-  exec(cmd, { timeout: 10000 }, // Process will time out if running for > 10 seconds.
+  var cp = exec(cmd, { timeout: 10000 }, // Process will time out if running for > 10 seconds.
     (error, stdout, stderr) => {
       if (error) {
-        return res.send(200, error)
+        return res.send(200, {"info":error, "process":cp.pid});
```

Comment:
Would it be possible to kill the process automatically here before sending the response? Then the client would not need to make another request.